radis/io/geisa.py

```diff
@@ -139,43 +144,78 @@ def fetch_geisa(
         engine=engine,
     )
 
-    # Get list of all expected local files for this database:
-    local_files, urlnames = ldb.get_filenames()
+    # Check if the database is registered in radis.json
+    local_files, urlnames = [], []
+    if ldb.is_registered():
+        entries = getDatabankEntries(ldb.name)
+        local_files, urlnames = entries["path"], entries["download_url"]
+
+    if ldb.is_registered() and not radis.config["ALLOW_OVERWRITE"]:
+        error = False
+        if cache == "regen" or not local_files:
+            error = True
+        files_to_check = local_files
+        if ldb.get_missing_files(files_to_check):
+            error = True
+        if error:
+            raise Exception(
+                'Changes are required to the local database, and hence updating the registered entry, but "ALLOW_OVERWRITE" is False. Set `radis.config["ALLOW_OVERWRITE"]=True` to allow the changes to be made and config file to be automatically updated accordingly.'
+            )
 
     # Delete files if needed:
-
     if cache == "regen":
         ldb.remove_local_files(local_files)
     ldb.check_deprecated_files(
         ldb.get_existing_files(local_files),
         auto_remove=True if cache != "force" else False,
     )
 
-    # Get missing files
-    download_files = ldb.get_missing_files(local_files)
+    get_main_files = True
 
-    # Download files
-    if len(download_files) > 0:
-        if urlnames is None:
-            urlnames = ldb.fetch_urlnames()
+    if len(local_files) > 1 or len(urlnames) > 1:
```

Comment:
```suggestion
    if len(local_files) != 1:
```
or is there something I'm missing?

Reply:
I followed Kurucz, but I think I'll change this on both now

Reply:
With this approach, we start with an empty array for local files, then search for them in the entries. If the db is not registered, we retrieve them using get_possible_files in the download section. So using if len(local_files) != 1 will raise an error if there are no entries, whereas it should only raise an error if there is more than one file or URI in radis.json.

Reply:
Sorry my bad